src/Microsoft.Azure.CosmosRepository/Builders/PatchOperationBuilder.cs

```diff
@@ -22,20 +22,31 @@ internal class PatchOperationBuilder<TItem> : IPatchOperationBuilder<TItem> wher
 
     public IPatchOperationBuilder<TItem> Replace<TValue>(Expression<Func<TItem, TValue>> expression, TValue? value)
     {
-        PropertyInfo property = expression.GetPropertyInfo();
-        var propertyToReplace = GetPropertyToReplace(property);
-        _rawPatchOperations.Add(new InternalPatchOperation(property, value, PatchOperationType.Replace));
+        IEnumerable<PropertyInfo> propertyInfos = expression.GetPropertyInfos();
+        var propertyToReplace = GetPropertyToReplace(propertyInfos);
+
+        _rawPatchOperations.Add(new InternalPatchOperation(propertyInfos.ToArray(), value, PatchOperationType.Replace));
         _patchOperations.Add(PatchOperation.Replace($"/{propertyToReplace}", value));
+
         return this;
     }
 
-    private string GetPropertyToReplace(MemberInfo propertyInfo)
```

Comment:
I think for backward compatibility we need to leave the old signature, and just have it call into the new one.

Something like this:

```csharp
private string GetPropertyToReplace(MemberInfo propertyInfo) => 
    GetPropertyToReplace([propertyInfo]);

Reply:
Done.